### PR TITLE
Refactored localize to use new l10n API

### PR DESCRIPTION
**Pull request by alexyaang**
This PR addresses issue #3693 

The changes replace `vscode-nls` with the vscode localization (`l10n`) API as introduced in https://code.visualstudio.com/updates/v1_72#_localization-as-part-of-the-api.

### CODE REVIEW

File: src/commands/registries/azure/deleteAzureRegistry.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import { DialogResponses, IActionContext } from '@microsoft/vscode-azext-utils';
 import { ProgressLocation, window } from 'vscode';
 import { ext } from '../../../extensionVariables';
-import { localize } from "../../../localize";
+import { l10n } from 'vscode';
```
Comment (danegsta) [inline src/commands/registries/azure/deleteAzureRegistry.ts:9]: There's an existing `vscode` import on line 7; when adding an import, double check that we're not already importing from the module. If there's an existing import, update it to add any additional exports you need to import.

**Review by danegsta — CHANGES_REQUESTED** [review]
The PR looks good for switching to the new API; only changes would be cleaning up the imports to avoid importing `vscode` multiple times (and to try to keep the import sorting consistent for easier readability).

File: src/commands/compose/compose.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import { IActionContext, UserCancelledError } from '@microsoft/vscode-azext-utils';
 import * as vscode from 'vscode';
 import { ext } from '../../extensionVariables';
-import { localize } from "../../localize";
+import { l10n } from 'vscode';
```
[the remark this answers is not in the document]
Comment (alexyaang) [reply src/commands/compose/compose.ts:9]: That makes sense, I'll make the change for all similar files.

File: src/commands/registries/azure/deleteAzureRegistry.ts (hunk 1 above)
Comment (alexyaang) [reply src/commands/registries/azure/deleteAzureRegistry.ts]: Just pushed the changes to reduce redundant imports, organize imports on files I touched. Let me know if there's anything else I should pay attention to!

File: package.json [hunk 3]
```diff
@@ -2891,7 +2891,6 @@
         "tar": "^6.1.11",
         "tree-kill": "^1.2.2",
         "vscode-languageclient": "^8.0.2",
-        "vscode-nls": "^5.0.0",
```
Comment (bwateratmsft) [inline package.json:2894]: I can't put the comment in the right location, but we need to update `@types/vscode` to 1.75.0, and also update the VSCode engine dependency to match.

Comment (alexyaang) [reply package.json:2894]: I see. Good catch!

**Review by bwateratmsft — APPROVED** [review]

**Review by danegsta — APPROVED** [review]

**Pull request merged**